internal/archive/archive.go

```diff
@@ -21,15 +21,15 @@ func init() {
 	NormalizedDateTime = time.Date(1980, time.January, 1, 0, 0, 1, 0, time.UTC)
 }
 
-func ReadDirAsTar(srcDir, basePath string, uid, gid int, mode int64, normalizeModTime bool) io.ReadCloser {
+func ReadDirAsTar(srcDir, basePath string, uid, gid int, mode int64, normalizeModTime bool, filter func(string) bool) io.ReadCloser {
```

Comment:
in this file, for readability, a type here would look nicer. Please we can document it.